app/controllers/badges_controller.rb

```diff
@@ -17,6 +17,29 @@ def show
 	def require
 	end
 
+  def new
+    if user_signed_in? && current_user.is_organizer
+      @badge = Badge.new
+    else
+      redirect_to new_user_session_path
+    end
+  end
+
+  def create
+    image                       = badge_params[:logo_url]
+    params[:badge][:logo_url]   = logo_url(image)
+    params[:badge][:edition_id] = @edition.id
```

Comment:
Is it possible to extract these lines to a different method? One that gives you already the result of `badge_params` and these changes you are making? The flow here is really hard to follow.